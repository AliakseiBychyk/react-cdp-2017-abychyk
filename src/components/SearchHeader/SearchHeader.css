.searchHeader {
  text-align: left;
  display: block;
  height: 200px;
  padding: 30px;
  margin: auto;
  background-color: slategray;
}

.searchByButtons {
  display: inline-block;
  padding: 15px;
}

.searchButton {
  float: auto;
  position: relative;
  left: 20%;
}