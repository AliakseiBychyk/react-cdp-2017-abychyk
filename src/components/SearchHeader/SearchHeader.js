import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import { Form, FormGroup, FormControl, ControlLabel, Button } from 'react-bootstrap'
import styles from './SearchHeader.css'

class SearchHeader extends Component {
  static defaultProps = {
    criterion: 'director',
    query: ''
  }

  constructor(props) {
    super(props)

    this.state = {
      criterion: props.criterion,
      query: decodeURIComponent(props.defaultQuery),
      fireRedirect: false
    }
  }

  setCriterion = () => {
    let criterion = this.state.criterion === 'director'
      ? 'title'
      : 'director'
    this.setState({ criterion })
  }

  setInputValue = (event) => {
    this.setState({ query: encodeURIComponent(event.target.value.trim()) })
  }

  submitForm = (e) => {
    e.preventDefault()
    this.props.redirectTo(`/search/${this.state.query}?criterion=${this.state.criterion}`) 
  }
  
  render() {
    const { fireRedirect } = this.state
    
    return (
      <div className={styles.searchHeader}>
        <Form onSubmit={this.submitForm} >       
            <ControlLabel>FIND YOUR MOVIE</ControlLabel>
            <FormControl
              type="text"
              name="title"
              placeholder="Quentin Tarantino"
              defaultValue={this.state.query}
              onChange={this.setInputValue}
            />
            <div className={styles.searchByButtons}>
              <ControlLabel>SEARCH BY</ControlLabel>
              
              <Button
                onClick={this.setCriterion}
                bsStyle={(this.state.criterion === 'title') ? 'primary' : 'default'}
              >TITLE</Button>
              
              <Button
                onClick={this.setCriterion}
                bsStyle={(this.state.criterion === 'director') ? 'primary' : 'default'}
              >DIRECTOR</Button>          
                
              <Button
                type="submit"
                className={styles.searchButton}
              >SEARCH</Button>
            </div>
        </Form>  
      </div>
    )
  }  
}

SearchHeader.propTypes = {
  criterion: PropTypes.string,
  reditectTo: PropTypes.func
}

export default SearchHeader