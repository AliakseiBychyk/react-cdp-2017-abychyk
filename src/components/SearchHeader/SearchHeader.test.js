import React from 'react'
import { shallow } from 'enzyme'
import SearchHeader from './SearchHeader'

let wrapper

beforeEach(() => {
  wrapper = shallow(<SearchHeader />)
});

test('should render SearchHeader correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should set query on input change', () => {
  const value = 'Quentin Tarantino'
  wrapper.find('FormControl').at(0).simulate('change', {
    target: { value }
  })
  expect(wrapper.state('query')).toBe(value)
})

test('should set criterion on button click', () => {
  wrapper.find('Button').at(0).simulate('click')
  expect(wrapper.state('criterion')).toBe('movie')
  wrapper.find('Button').at(1).simulate('click')
  expect(wrapper.state('criterion')).toBe('person')
})

test('should fireRedirect on button click', () => {
  const query = wrapper.state('query')
  const criterion = wrapper.state('criterion')
  wrapper.find('Form').simulate('submit', {
    preventDefault: () => { }
  }, )
  expect(wrapper.props('redirectTo')).toEqual(`/search/${query}?criterion=${criterion}`)
})
