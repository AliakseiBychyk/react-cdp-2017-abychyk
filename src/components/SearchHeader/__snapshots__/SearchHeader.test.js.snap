// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchHeader correctly 1`] = `
<div
  className="searchHeader"
>
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={false}
    inline={false}
    onSubmit={[Function]}
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      FIND YOUR MOVIE
    </ControlLabel>
    <FormControl
      bsClass="form-control"
      componentClass="input"
      name="title"
      onChange={[Function]}
      placeholder="Quentin Tarantino"
      type="text"
    />
    <div
      className="searchByButtons"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        SEARCH BY
      </ControlLabel>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        onClick={[Function]}
      >
        MOVIE
      </Button>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="primary"
        disabled={false}
        onClick={[Function]}
      >
        DIRECTOR
      </Button>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="searchButton"
        disabled={false}
        type="submit"
      >
        SEARCH
      </Button>
    </div>
  </Form>
</div>
`;
